src/config/settings.rs

```diff
@@ -14,6 +15,92 @@ pub enum Setting {
     Map(Vec<String>),
 }
 
+impl Setting {
+    pub fn to_bool(&self) -> bool {
+        if ! matches!(self, Setting::Bool(_)) {
+            warn!("setting is not a boolean");
+        }
+
+        match self {
```

Comment:
even if we have a bool-setting, it may be possible to read it as another type. It's not recommended in most cases, but i can see that it sometimes can be. However, we will always log a warning when we do so.